### PR TITLE
[DOCS] Add synthetic _source to 8.4 release highlights

### PR DESCRIPTION
**Pull request by abdonpijpelink**
This PR adds a section about synthetic _source to the 8.4 release highlights.

### CODE REVIEW

**Comment by abdonpijpelink** [comment]
@elasticmachine update branch

File: docs/reference/release-notes/highlights.asciidoc [hunk 1]
```diff
@@ -106,6 +106,22 @@ search features like queries and aggregations.
...
+When Elasticsearch ingests documents, it creates several data structures that
+enable querying and aggregating the data. One of those data structures is called
+`_source`.  `_source` holds the fields of every document. It uses a lot of space
+and is used for specific needs, like reindexing and presenting the document in
+Kibana Discover. 
+
+This release introduces a new feature in technical preview called synthetic
+`_source`. For specific data types, synthetic `_source` can recreate the
+`_source` from doc values when needed. This option significantly reduces the
+index size.
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/highlights.asciidoc:121]: I'd love to hear more about the benefits of this feature, such as when and where folks would typically use it. It reduces the index size, but perhaps we can provide information about when that's necessary? Something like:

>Synthetic `_source` significantly reduces the index size by recreating the `_source` from doc values, and is useful for when you don't need to store everything in `_source`. While doc values don't preserve the original structure of the `_source`, they're enough to recreate the `_source` and still allow for capabilities like reindexing.

I'm taking some liberties based on a quick review of the related PR, so don't accept that suggestion verbatim 😆

**Review by lockewritesdocs — COMMENTED** [review]
Thanks @abdonpijpelink -- I left some suggestions about providing more explanation about the benefits of this feature.

File: docs/reference/release-notes/highlights.asciidoc (hunk 1 above)
Comment (abdonpijpelink) [reply docs/reference/release-notes/highlights.asciidoc]: Thanks @lockewritesdocs . I've pushed a change that incorporates your feedback.

File: docs/reference/release-notes/highlights.asciidoc [hunk 2]
```diff
@@ -106,6 +106,25 @@ search features like queries and aggregations.
 
 {es-pull}88694[#88694]
 
+[discrete]
+[[synthetic_source_technical_preview]]
+=== Synthetic `_source`
+preview::[]
+When Elasticsearch ingests documents, it creates several data structures that
+enable querying and aggregating the data. One of those data structures is called
+`_source`.  `_source` holds the fields of every document. It uses a lot of space
```
Comment (nik9000) [inline docs/reference/release-notes/highlights.asciidoc:115]: `_source` is actually for "fetching" the document, not really "querying" or "aggregating"

File: docs/reference/release-notes/highlights.asciidoc [hunk 3]
```diff
@@ -106,6 +106,25 @@ search features like queries and aggregations.
...
+and is used for specific needs, like reindexing and presenting the document in
+Kibana Discover.
```
Comment (nik9000) [inline docs/reference/release-notes/highlights.asciidoc:117]: You can be more precise here - something like "it's used when we need to return the exact document that you sent us, like reindex, _update, and Kibana's discover".

File: docs/reference/release-notes/highlights.asciidoc [hunk 4]
```diff
@@ -106,6 +106,25 @@ search features like queries and aggregations.
...
+`_source`.  `_source` holds the fields of every document. It uses a lot of space
+and is used for specific needs, like reindexing and presenting the document in
+Kibana Discover. 
+
+This release introduces a new feature in technical preview called synthetic
+`_source`. Synthetic `_source` can significantly reduce the index size by, for
+specific data types, recreating the `_source` from doc values. This is useful
+when you don't need to preserve the original structure of the `_source`. While
+doc values don't preserve the original structure of the `_source`, synthetic
+`_source` still allows for capabilities like reindexing.
```
Comment (nik9000) [inline docs/reference/release-notes/highlights.asciidoc:124]: I think the critical thing you don't mention is that we need to store the doc values anyway. I tend to explain synthetic _source like this:

Elasticsearch can store 3 things, which have three primary use cases: a search index for finding the documents, a column store called doc_values for aggregating the documents, and _source for returning the precise document you sent us. A lot of people don't need to get the precise document that they send us back, but the whole ecosystem around Elasticsearch relies on being able to get it. So synthetic _source let's us rebuild a look-alike on the fly. It doesn't preserve the precise order of things, but that's *fine* for lots of people. Like I said, lots of people never even look at `_source`. It does take a little longer to fetch, but that performance doesn't effect aggregations at all, which tend to be the bigger performance bottleneck most folks face.

Comment (abdonpijpelink) [reply docs/reference/release-notes/highlights.asciidoc]: @nik9000 I've pushed a change that includes the changes you've suggested. Let me know what you think.

File: docs/reference/release-notes/highlights.asciidoc [hunk 5]
```diff
@@ -22,10 +22,114 @@ Other versions:
...
 // should be featured in the Stack Installation and Upgrade Guide:
 // tag::notable-highlights[]
-// [discrete]
-// === Heading
-//
-// Description.
+
+[discrete]
+[[speed_up_filters_range_date_histogram_aggs]]
+=== Speed up filters/range/date_histogram aggs
```
[the remark this answers is not in the document]
Comment (abdonpijpelink) [reply docs/reference/release-notes/highlights.asciidoc:28]: I think this came from merging main. This text is already there in the original [doc](https://github.com/elastic/elasticsearch/blob/8.4/docs/reference/release-notes/highlights.asciidoc).

This PR is targeting the wrong branch. It should target 8.4. I've submitted https://github.com/elastic/elasticsearch/pull/90269 . We can continue the conversation there.

**Comment by abdonpijpelink** [comment]
Closing this in favor of https://github.com/elastic/elasticsearch/pull/90269 which targets the correct branch.

**Pull request closed without merging**